backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/sort/nested/impl/LuceneNestedFieldComparator.java

```diff
@@ -18,4 +18,21 @@
 	public void setNestedDocumentMap(Map<Integer, Set<Integer>> nestedDocumentMap) {
 		this.nestedDocumentMap = nestedDocumentMap;
 	}
+
+	public Integer getNestedDocument(Integer rootDocument) {
+		// on the first call the nestedDocumentMap is not set
+		if ( nestedDocumentMap == null ) {
+			return rootDocument;
+		}
+
+		// it is possible that a document does not have a value set for the nested field
+		Set<Integer> nestedDocs = nestedDocumentMap.get( rootDocument );
+		if ( nestedDocs == null ) {
+			return rootDocument;
+		}
+
+		// TODO at the moment we do not support multi value sorting on nested document
+		// using one of them in case
```

Comment:
I think you could use a `Map<Integer, Integer>`, in that case. I suspect that, even when we'll support multi-value sorting, we'll do the nested document selection upfront, not in here, so we'll never need more than one document ID here.
That would spare us one `Set` instantiation per document, which may or may not be a big deal, but I suspect it will.

Reply:
I thought about it and I think you definitely right. Given a single nested document path we can have at most one nested document for each root document.
I think that we have a set here, just because I've reused the same nested query we already had for nested projection.

Reply:
Done in a different commit. We could revert it, if we were to realize that we were wrong.